### PR TITLE
v2.10.0 - upgrade to React 17

### PR DESCRIPTION
**Pull request by shaunanoordin**
## PR Overview

This PR upgrades the peer dependencies to React 17, and bumps the package to version 2.10.0

This repo may be deprecated, but it's still a dependency on Panoptes-Front-End. Its usage of React 16 is causing serious issues with the build process on Node 16.15. See zooniverse/Panoptes-Front-End#6155

### Status

- WIP. Changes should work in theory, but I'm testing a local PFE build with this.
- I'm trying to see how far I can update the other dependencies as well, for security's sake.

### CODE REVIEW

**Comment by shaunanoordin** [comment]
Oh thank gods the other dependencies could be updated with no issues. `npm audit fix` is now clear.

File: package.json [hunk 1]
```diff
@@ -1,14 +1,14 @@
 {
   "name": "modal-form",
-  "version": "2.9.1",
+  "version": "2.10.0",
   "devDependencies": {},
   "peerDependencies": {
-    "react": "^16.2.0",
-    "react-dom": "^16.2.0"
+    "react": "17.x",
+    "react-dom": "^17.x"
```
Comment (eatyourgreens) [inline package.json:7]: The peer dependencies should include React 16 and React 17.  See Markdownz for an example.

**Review by eatyourgreens — CHANGES_REQUESTED** [review]
Peer dependencies should be React greater than 16.2.

**Comment by shaunanoordin** [comment]
I'm closing PR this since it's no longer too urgent - we have a workaround for PFE PR 6155, which is to use `--legacy-peer-deps`.

**Pull request closed without merging**